scalding-core/src/main/scala/com/twitter/scalding/Operations.scala

```diff
@@ -491,13 +497,23 @@ package com.twitter.scalding {
         .asScala
         .map(_.getObject(0).asInstanceOf[V])
 
+      val caches = values.toList
```

Comment:
so, we can't do this (because it would force everything to memory), but what about something like this:
```scala
class CountingIterator[T](wraps: Iterator[T]) extends Iterator[T] {
  private[this] var nextCalls = 0
  def hasNext = wraps.hasNext
  def next = { nextCalls += 1; wraps.next }
  def seen: Int = nextCalls
}
```
then we could wrap `values` with this, then call `.seen` at the end to see how many values went in?